### PR TITLE
Travis CI support

### PR DESCRIPTION
**Pull request by philipmw**
Allow TravisCI to run specs for all Rubies we want to support.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+language: ruby
+rvm:
+  - 1.8.7
```
Comment (bhuga) [inline .travis.yml]: The README is old and this isn't a goal anymore.

**Comment by bhuga** [comment]
Looks like this fails without the gemfile changes: https://travis-ci.org/philipmw/promising-future/jobs/61394756

I just merged the bundler stuff. Can you rebase/merge master and try again?

File: .travis.yml (hunk 1 above)
Comment (philipmw) [inline .travis.yml]: But it still works! :) I am motivated to add all Rubies that currently work, even if it's not a goal to continue supporting them, for two reasons:
1) Customers can feel confident using this gem on whatever interpreter they have;
2) We make an informed decision about whether to accept a pull request / make a change knowing that it breaks an interpreter.

**Comment by philipmw** [comment]
Rebased. Also updated the README.

File: .travis.yml (hunk 1 above)
Comment (bhuga) [inline .travis.yml]: Fair enough. We can let it hang around for now.

**Comment by bhuga** [comment]
Sadness: https://travis-ci.org/philipmw/promising-future/jobs/61494611

**Comment by philipmw** [comment]
Yes this has uncovered some existing issues on some Rubies. I think it's ok to have the build failing for now, and we should fix these issues in separate PRs.

**Comment by bhuga** [comment]
> some Rubies.

Just the one :wink:

>  I think it's ok to have the build failing for now, and we should fix these issues in separate PRs.

This makes me really nervous, because I won't have time to mess with jruby anytime soon. We have to do another PR anyway, so let's remove the failure from the matrix and re-add it with the PR that fixes it. Soon as we do that this is :shipit:

**Comment by philipmw** [comment]
Unfortunately there are intermittent failures on Rubinius as well: https://travis-ci.org/philipmw/promising-future/jobs/61394872

I will remove all but MRI Rubies for now.

**Comment by philipmw** [comment]
updated.

**Comment by bhuga** [comment]
Awesome. Thanks! I'd want to get to this for a long time.

you re awesome

**Comment by philipmw** [comment]
Thanks. Now that this is merged, I discovered that I cannot enable Travis CI for this repo because I don't have admin rights. Can you enable it for us or give me the needed bit?

**Comment by bhuga** [comment]
> Can you enable it for us

Already done!

**Pull request merged**